doc/voq/voq_hld.md

```diff
@@ -0,0 +1,874 @@
+
+ ![](../../images/voq_hld/voq-bigpicture.png)
+
+Each LC above is a Single SONiC Linux System. Each LC may have one or more asics. Each asic is controlled by a separate instance of the SONiC Network Stack (lldpd, bgp, swss, syncd, redis, teamd, ..). No changes are proposed to any of the shared (by all the asics on the line card) SONiC components on the card. 
+
+The Control Card is its own separate SONiC Linux System. It does not run any of the SONiC Network protocol components (lldp, bgp, teamd ...). But it has an SWSS and SYNCD instance for control of the Switch Fabric asics.
+
+# 1 Requirements Overview
+## 1.1 Functional Requirements
+
+The VOQ feature implementation shall support the following
+1.  Distributed VOQ System.
+2.  IPv4 and IPv6 unicast routing across any ports in the system
+3.  Each switch in the system is controlled by a separate "asic instance" of SONiC.
+4.  Host IP reachability to/from the interface IP addresses of any SONiC instance in the system using any network port in the system.
+5.  Host IP reachability between the SONiC instances in the system over the datapath.
+6.  Routing protocol peering between SONiC instances over the datapath.
+7.  Static provisioning of System Ports in the VOQ System Database.
+8.  Dynamic discovery of Routing Interfaces and Neighbors on other asics via the VOQ System Database.
+9.  Automatic determination of Switch_Id for each asic (optional).
+
+## 1.2 Platform requirements
+Every ASIC in the system needs be assigned a unique Switch_ID. Each ASIC consumes as many consecutive switch_id values as it has cores. So the next valid switch_id assignment should skip ahead by a number equal to the number of cores in the asic. This can be done via an additional configuration attribute. Another option is atuomatic determination of sWwitch_ID. Vendor supplied platform specific components are aware of details (line card number, asic number with in the line card, number of cores for the asic etc...) which allow them to automatically calculate the Switch_ID for each asic. For this reason it is proposed the PMON could populate the Switch_ID for an asic into the APPL_DB.
+
+## 1.3 Configuration requirements
+Phase-1 of the Distributed VOQ System should support static configuration of the following
+1. Connection parameters for the Central VOQ System Database (IP address, Port)
+2. All the system ports in the entire system. The configuration of each system port should specify - system_port_id, switch_id, cored_index, core_port_index and speed.
+3. Maximum number of cores in the system.
+4. Switch_ID for each asic
+
+## 1.4 Orchagent requirements
+### Switch Creation
+ - Should create the switch with the correct values for max_cores and switch_id
+ - Should create the switch with complete list of system port for all the system ports during initialization
+
+### PortsOrch:
+ - Should be SystemPorts aware
+ - Should initialize system ports list and create host interfaces for system ports
```

Comment:
Please update that host i/fs are created ONLY for local ports and not system ports.